kratos/python_scripts/read_materials_process.py

```diff
@@ -26,47 +26,71 @@ def __init__(self, Model, settings):
         
         print("finished reading materials")
         
-    def _GetItemFromModule(self,my_string):
-        """Return the python object named by the string argument.
+    def _GetVariable(self,my_string):
+        """Return the python object of a Variable named by the string argument.
+
+        Examples:
+        variable = self._GetVariable("VELOCITY")
+        variable = self._GetVariable("KratosMultiphysics.VELOCITY")
+        variable = self._GetVariable("SUBSCALE_PRESSURE")
+        variable = self._GetVariable("FluidDynamicsApplication.SUBSCALE_PRESSURE")
+        variable = self._GetVariable("KratosMultiphysics.FluidDynamicsApplication.SUBSCALE_PRESSURE")
+        """
+        splitted = my_string.split(".")
+
+        if len(splitted) == 0:
+            raise Exception("Something wrong. Trying to split the string " + my_string)
+        if len(splitted) > 3:
+            raise Exception("Something wrong. String " + my_string + " has too many arguments")
+
+        return KratosMultiphysics.KratosGlobals.GetVariable(splitted[-1]) # This also checks if the application has been imported
+
+    def _GetConstitutiveLaw(self,my_string):
+        """Return the python object of a Constitutive Law named by the string argument.
 
         Example:
-        constitutive_law = self._GetItemFromModule('KratosMultiphysics.StructuralMechanicsApplication.LinearElastic3DLaw')
+        constitutive_law = self._GetConstitutiveLaw('KratosMultiphysics.StructuralMechanicsApplication.LinearElastic3DLaw')
         model_part.GetProperties(prop_id).SetValue(CONSTITUTIVE_LAW, constitutive_law)
         """
         splitted = my_string.split(".")
         if(len(splitted) == 0):
             raise Exception("something wrong. Trying to split the string "+my_string)
         if(len(splitted) == 1):
-            return eval(my_string)
+            raise Exception("Please also provide the name of the application of constitutive law "+my_string)
+
+        constitutive_law_name = splitted[-1]
+        module_name = splitted[-2]
+
+        if module_name == "KratosMultiphysics":
+            return getattr(KratosMultiphysics, constitutive_law_name) 
         else:
-            module_name = ""
-            for i in range(len(splitted)-1):
-                module_name += splitted[i] 
-                if i != len(splitted)-2:
-                    module_name += "."
-
-            module = importlib.import_module(module_name)
-            return getattr(module,splitted[-1]) 
-             
+            application_name = "Kratos" + module_name
+            if application_name not in KratosMultiphysics.KratosGlobals.RequestedApplications:
+                raise ImportError(module_name + " is not imported!")
+            module1 = KratosMultiphysics.KratosGlobals.RequestedApplications[application_name]
+            module2 = sys.modules[application_name]
+            
+            return getattr(module2, constitutive_law_name) 
```

Comment:
@KratosMultiphysics/technical-committee if I use `module1` here (aka `KratosMultiphysics.KratosGlobals.RequestedApplications[application_name]`), it is not working.
Do you have an idea why?

Reply:
honestly no. I also had a lot of troubles with this.